### PR TITLE
[java] Cleanup isAllowed method as it can be much more simple

### PR DESCRIPTION
**Pull request by hazendaz**

### CODE REVIEW

File: license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java [hunk 1]
```diff
@@ -51,16 +51,10 @@ public boolean getPredicate() {
...
      */
     public boolean isAllowed(final boolean matched) {
-      if (matched && allowed) {
-        return true;
-      } else if (!matched && !allowed) {
-        return true;
-      } else {
-        return false;
-      }
+      return (matched && allowed) || (!matched && !allowed);
```
Comment (mathieucarbou) [inline license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java:57]: `return !(matched ^ allowed)`

**Review by mathieucarbou — CHANGES_REQUESTED** [review]

File: license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java (hunk 1 above)
Comment (hazendaz) [reply license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java:57]: think that does not apply given how the checks are but Eclipse is suggesting far, far easier -> 'returned matched == allowed'.

Comment (hazendaz) [reply license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java:57]: updating PR with suggestion from eclipse better look at.

Comment (mathieucarbou) [reply license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/dependencies/LicensePolicy.java]: lol indeed ;-) `false == false`

**Review by mathieucarbou — APPROVED** [review]

**Pull request merged**